File: src/test/java/org/veri/be/unit/member/MemberConverterTest.java
package org.veri.be.unit.member;

import org.junit.jupiter.api.DisplayName;
import org.junit.jupiter.api.Nested;
import org.junit.jupiter.api.Test;
import org.veri.be.domain.member.converter.MemberConverter;
import org.veri.be.domain.member.dto.MemberResponse;
import org.veri.be.domain.member.entity.Member;
import org.veri.be.domain.member.entity.enums.ProviderType;

import java.lang.reflect.Constructor;
import java.lang.reflect.Modifier;

import static org.assertj.core.api.Assertions.assertThat;

class MemberConverterTest {

    @Nested
    @DisplayName("toMemberInfoResponse")
    class ToMemberInfoResponse {

        @Test
        @DisplayName("멤버 정보 응답으로 변환한다")
        void convertsToResponse() {
            Member member = member();

            MemberResponse.MemberInfoResponse response = MemberConverter.toMemberInfoResponse(member, 2, 3);

            assertThat(response.getEmail()).isEqualTo(member.getEmail());
            assertThat(response.getNickname()).isEqualTo(member.getNickname());
            assertThat(response.getImage()).isEqualTo(member.getProfileImageUrl());
            assertThat(response.getNumOfReadBook()).isEqualTo(2);
            assertThat(response.getNumOfCard()).isEqualTo(3);
        }
    }

    private Member member() {
        return Member.builder()
                .id(1L)
                .email("member@test.com")
                .nickname("member")
                .profileImageUrl("https://example.com/profile.png")
                .providerId("provider-1")
                .providerType(ProviderType.KAKAO)
                .build();
    }

    @Test
    @DisplayName("생성자는 외부에서 호출할 수 없도록 private이어야 한다")
    void constructorShouldBePrivate() throws NoSuchMethodException {
        // given
        Constructor<MemberConverter> constructor = MemberConverter.class.getDeclaredConstructor();

        // when & then
        assertThat(Modifier.isPrivate(constructor.getModifiers())).isTrue();
    }
}

File: tests/src/test/kotlin/org/veri/be/unit/member/MemberConverterTest.kt
package org.veri.be.unit.member

import org.assertj.core.api.Assertions.assertThat
import org.junit.jupiter.api.DisplayName
import org.junit.jupiter.api.Nested
import org.junit.jupiter.api.Test
import org.veri.be.domain.member.dto.MemberResponse
import org.veri.be.support.fixture.MemberFixture

class MemberConverterTest {

    @Nested
    @DisplayName("toMemberInfoResponse")
    inner class ToMemberInfoResponse {

        @Test
        @DisplayName("멤버 정보 응답으로 변환하면 → 결과를 반환한다")
        fun convertsToResponse() {
            val member = member()

            val response: MemberResponse.MemberInfoResponse =
                MemberResponse.MemberInfoResponse.from(member, 2, 3)

            assertThat(response.email).isEqualTo(member.email)
            assertThat(response.nickname).isEqualTo(member.nickname)
            assertThat(response.image).isEqualTo(member.profileImageUrl)
            assertThat(response.numOfReadBook).isEqualTo(2)
            assertThat(response.numOfCard).isEqualTo(3)
        }
    }

    private fun member(): org.veri.be.domain.member.entity.Member {
        return MemberFixture.aMember().id(1L).nickname("member").build()
    }

}
